src/cascadia/Remoting/Monarch.cpp

```diff
@@ -147,6 +147,56 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
         }
     }
 
+    uint64_t Monarch::_lookupPeasantIdForName(const winrt::hstring& name)
+    {
+        if (name.empty())
+        {
+            return 0;
+        }
+
+        std::vector<uint64_t> peasantsToErase{};
+        uint64_t result = 0;
+        for (const auto& [id, p] : _peasants)
+        {
+            try
+            {
+                auto otherName = p.WindowName();
+                if (otherName == name)
+                {
+                    result = id;
+                    break;
+                }
+            }
+            catch (...)
+            {
+                LOG_CAUGHT_EXCEPTION();
+                // Normally, we'd just erase the peasant here. However, we can't
+                // erase from the map while we're iterating over it like this.
+                // Instead, pull a good ole Java and collect this id for removal
+                // later.
+                peasantsToErase.push_back(id);
+
+                // // Remove the peasant from the list of peasants

```

Comment:
Trim leftover code since you're not doing it this way anymore.

Reply:
agree.